### ISSUE TITLE
Some fights pages return an internal error

### ISSUE THREAD

**Issue by craaig4** [GaryA/antlog#91]
Hi, sorry if this is incorrectly formatted. Not particularly used to Github.

Receiving internal error #8 when trying to access pages 3-6 of the AWS54 Fights section. Thought it's worth flagging up.

**Comment by GaryA** [comment]
Thanks for reporting this issue. I'll look into it, but initially I'm not sure why only some pages should give an error...On 18 Mar 2018 9:35 p.m., Craig <notifications@github.com> wrote:Hi, sorry if this is incorrectly formatted. Not particularly used to Github.
Receiving internal error #8 when trying to access pages 3-6 of the AWS54 Fights section. Thought it's worth flagging up.

**Comment by GaryA** [comment]
This issue is caused by missing records in the `entrants` table.
Only AWS53 and AWS54 are affected.
Entrants 497 (group 6), plus 442 and 468 (group 7) are missing from AWS53 (event 7)
Entrant 605 is missing from AWS54 (event 9)
There are other entrants "missing" but this is OK because they don't appear in the `fights` table, so they have been deleted, probably fleas or nanos that were signed up for ant events then withdrawn.

**Comment by GaryA** [comment]
Found the root cause.

Some of the entrants are being exported from the local version of Antlog with the status field set to -1 even though they have taken part in the event and should end up with status = 0, 1 or 2.
At the end of the SQL file all entrants to complete events are deleted if their status is -1 (did not compete).

So the question is, why does the status end up at -1 for entrants that have competed?
This only seems to have happened since AWS53 (November 2017)

**Comment by craaig4** [comment]
Something relating to Chris' virtual machine install perhaps? As always, I have no idea what I'm talking about but try to offer vague suggestions ;)

**Comment by GaryA** [comment]
I'd hesitate to suggest it's Chris's fault, there were probably changes made to Antlog between AWS50 when it was last used without issue and AWS53 when the problem first appeared.
I've now fixed the Antlog database, I managed to recover the missing records from old copies of the database.

**Comment by craaig4** [comment]
Aha! That has actually proved related to the reason I was looking into the fight list...
Entrant 605 is The violation of our most basic human rights

Group 7 Round 3, Loser's Bracket, Pingu beats TVOOMBHR.
However, strangely, in Round 4 TVOOMBHR was drawn instead of Pingu.
That Round 4 fight didn't actually happen as the team had rightly assumed all their robots were out and had left. Seems some fault had put them through and Pingu out despite Pingu's win.

**Comment by GaryA** [comment]
Yes I noticed that, and I don't ywt know why it happened, but it might explain why the entrant record was exported with status = -1. The status is used to track the "lives" in double elimination. It starts at 2 and decrements when you lose. Everyone should end up at zero except the winner. Since TVOOMBHR lost 3 fights its status would get to -1.
However this doesn't explain what happened at AWS53 as 3 entrants ended up with status = -1 but all of their fights & results look correct.

**Comment by GaryA** [comment]
@cjsoftuk I'm struggling to find and reproduce the root cause of the error. I assume it was caused by a result being changed, but the entries in the log don't quite tie up with what normally happens and when I reply the queries in the log I still don't get to the erroneous state.
I will try recreating the event and logging all the results up to where the wrong entry was changed, but it's a lot of boring work.
Every time I've tested changing a result it has worked perfectly. It's a necessary feature as there will always be finger trouble, and I can't find any issue with the logic so far.

**Comment by cjsoftuk** [comment]
Thanks Gary - I agree that it's boring to go through that - am happy to do the testing there if needed. I'm intending to spend all of next weekend doing prep for AWS55 so if we don't have any progress by then I will try and do some more digging.

**Comment by GaryA** [comment]
The problem is trying to exactly recreate the fights table as it was at the start of the event, after the draw which is probably going to involve a lot of exporting and splicing SQL files. Entering the results is tedious but not the most onerous part.At least I've fixed the disc-eating bug :)On 21 May 2018 7:20 p.m., cjsoftuk <notifications@github.com> wrote:Thanks Gary - I agree that it's boring to go through that - am happy to do the testing there if needed. I'm intending to spend all of next weekend doing prep for AWS55 so if we don't have any progress by then I will try and do some more digging.

**Comment by GaryA** [comment]
Rebuilt database as it was at the start of AWS54, with the draw done but no fights run.
Ran all fights, including Group 7 Round 3, Loser's Bracket, Pingu beats TVOOMBHR, but recorded as TVOOMBHR winning.
Ran a couple more fights then went back and corrected Group 7 Round 3 Loser's Bracket to set Pingu as winner.
Continued with fights and Group 7 Round 4 was correct.
At the end all the data was correct, all entries had status = 0 except winner (status = 2). No entries with status = -1

Don't know how to reproduce the problem....

**Comment by cjsoftuk** [comment]
Very confused how that happened then (doubly odd since it didn't happen at AWS53 and there wasn't a lot between them code-wise).  I guess we'll just have to keep a close eye on it to see if it happens again.

**Comment by GaryA** [comment]
Something similar happened at AWS53 but the fights code was no different to previous events which were ok. I thought about PHP timing out due to different server settings but there were no errors in the logs. I'll add some code to detect if things are going wrong and fix them and log as much as possible about what might have happened.On 21 May 2018 11:08 p.m., cjsoftuk <notifications@github.com> wrote:Very confused how that happened then (doubly odd since it didn't happen at AWS53 and there wasn't a lot between them code-wise).  I guess we'll just have to keep a close eye on it to see if it happens again.
